### DISCUSSION TITLE
Scalability solution/path to distributed environment

### DISCUSSION THREAD

**Discussion by thomasguo** [duckdb/duckdb#2345] [Q&A]
I am aware that duckdb is primarily targeting at embedded environment.  That's great and very valuable.  However, I am wondering if there is a clear/easy path for duckdb to scale to a server/distributed  mode.  That way applications written for current embedded duckdb could easily adapt to a multi-user or multi-node environment.  As far as I know some examples are:

1)  H2 embedded mode --> H2 server mode.
2)  monetdb/e --> monetdb standalone server.
3)  postgres --> greenplum or redshift.
...

**Comment by sahuguet** [comment]
What about embedding DuckDB in your favorite REST API server, e.g. Flask? You put a reverse proxy at the top to distribute the queries against various REST API nodes running DuckDB.

**Reply by hannes** [reply]
We already have a multi-threaded REST server in our repo, feel free to check it out.
